lib/vm.js

```diff
@@ -286,6 +287,72 @@ function runInThisContext(code, options) {
   return createScript(code, options).runInThisContext(options);
 }
 
+function compileFunction(code, params, options) {
+  const defaults = {
+    filename: '',
+    columnOffset: 0,
+    lineOffset: 0,
+    cachedData: undefined,
+    produceCachedData: false,
+    parsingContext: undefined
+  };
+
+  if (typeof code !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE('code', 'string', code);
+  }
+  if (params !== undefined) {
+    if (!Array.isArray(params)) {
+      throw new ERR_INVALID_ARG_TYPE('params', 'Array', params);
+    }
+    params.forEach((param, i) => {
+      if (typeof param !== 'string') {
+        throw new ERR_INVALID_ARG_TYPE(`params[${i}]`, 'string', param);
+      }
+    });
+  }
+
+  if (options !== undefined) {
+    for (const key in options) {
+      const type = typeof defaults[key];
+      const desiredType = (type !== 'undefined' ? type : 'object');
+      if (desiredType === 'object' && options[key] === null) {
+        throw new ERR_INVALID_ARG_TYPE(
+          `options.${key}`,
+          'non-null object',
+          options[key]
+        )
+      }
+      if (typeof options[key] !== desiredType) {
+        throw new ERR_INVALID_ARG_TYPE(
+          `options.${key}`,
+          desiredType,
+          options[key]
+        );
+      }
+    }
+  }
+
+  const {
+    filename,
+    columnOffset,
+    lineOffset,
+    cachedData,
```

Comment:
This is not type-checked.

Reply:
Could you elaborate? It *is* checked if `cachedData` is an `Object` or not, but it's not checked in it's a `Uint8Array`. Do you propose I add checking for the exact type? I think I *should* do that, but couldn't find instances of the same in the codebase.

Reply:
> Do you propose I add checking for the exact type?

Yes.

> I think I *should* do that, but couldn't find instances of the same in the codebase.

Because the entire (or at least most of) codebase is now free of Uint8Array-only functions. You could use [`isUint8Array()`](https://github.com/nodejs/node/blob/7951e6d26b2c2036a8d6dbb92b395f9e15d90e14/lib/internal/util/types.js#L28-L30) in `internal/util/types.js` for this purpose.